content/en/blog/2022/instrument-kafka-clients/index.md

```diff
@@ -0,0 +1,252 @@
+</dependency>
+```
+
+In this way, you have the minimum set up for enabling the tracing on your Kafka based application.
+
+### Setting up the OpenTelemetry instance
+
+The entire tracing instrumentation is handled by an `OpenTelemetry` instance in your code.
+It needs to be created and registered globally in order to be made available to the Kafka clients instrumentation library.
+
+This can be done in two different ways:
+
+* using an SDK extension for environment-based auto-configuration.
+* using SDK builders classes for programmatic configuration.
+
+#### Using the SDK auto-configuration
+
+It is possible to configure a global `OpenTelemetry` instance by using environment variables thanks to the SDK extension for auto-configuration, enabled by adding the following dependency to your application.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
+</dependency>
+```
+
+When the Kafka clients instrumentation library is used, it checks if there is an `OpenTelemetry` instance already created and registered.
+If it is not, the library code checks if the SDK auto-configure module is in the classpath and in this case initializes it for creating the `OpenTelemetry` instance automatically.
+The corresponding configuration happens through environment variables (or corresponding system properties).
+This is really a way to simplify the initialization of tracing. 
+
+The main environment variables to be set are the following:
+
+* `OTEL_SERVICE_NAME`: specify the logical service name. This is really useful when using a tracing UI (i.e. Jaeger UI) for showing data.
+* `OTEL_TRACES_EXPORTER`: the list of exporters to be used for tracing. For example, by using `jaeger` you also need to have the corresponding dependency in the application.
+
+Instead of using the above environment variables, it is also possible to use corresponding system properties to be set programmatically in the code or on the command line when launching the application
+They are `otel.service.name` and `otel.traces.exporter`.
+
+#### Using the SDK builders
+
+In order to build your own `OpenTelemetry` instance and not relying on the auto-configuration, it is possible to do so by using the SDK builders classes programmatically.
+The OpenTelemetry SDK dependency is needed in order to have such SDK builders available in your code.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-sdk</artifactId>
+</dependency>
+```
+
+The following code snippet sets the main attributes like the service name, then it configures the Jaeger exporter.
+Finally, it creates the `OpenTelemetry` instance and registers it globally so that it can be used by the Kafka clients instrumentation library.
+
+```java
+Resource resource = Resource.getDefault()
+        .merge(Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, "my-kafka-service")));
```

Comment:
above you ask people to use the environment variable / system property, now you're setting the SERVICE_NAME via code as well, that might confuse the user.

Reply:
This is the section related to not using the SDK auto-configuration (which relies on env vars as well) but it's about creating the `OpenTelemetry` instance programmatically from your own code.

Reply:
ah, gotcha